compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2092,6 +2092,11 @@ impl<'tcx> TyS<'tcx> {
         }
     }
 
+    #[inline]
+    pub fn is_ptr(&self) -> bool {
```

Comment:
This is a confusing name, given that methods like `is_mutable_ptr` also include reference types. I do not have a good idea though, beyond `is_raw_or_fn_ptr`...

Also please add the new method next to the existing `is_*_ptr` methods to keep them all together.

Reply:
Good point. A search for `is_ptr` only turned up `is_ptr_sized_integral` below. I've just removed this and used `is_unsafe_ptr || is_fnt_ptr`.

Reply:
Yeah it is probably not worth having a helper for a single call site.